pkg/sentry/fs/fsutil/cache_reclaim.go

```diff
@@ -0,0 +1,350 @@
+package fsutil
+
+import (
+        "sync"
+        "sync/atomic"
+        "time"
+        "runtime"
+
+        "gvisor.dev/gvisor/pkg/ilist"
+        "gvisor.dev/gvisor/pkg/log"
+        "gvisor.dev/gvisor/pkg/sentry/context"
+        "gvisor.dev/gvisor/pkg/sentry/memmap"
+        "gvisor.dev/gvisor/pkg/sentry/pgalloc"
+        "gvisor.dev/gvisor/pkg/sentry/platform"
+        "gvisor.dev/gvisor/pkg/sentry/usermem"
+)
+
+const (
+        LRU_ACTIVE_LIST = 0
+        LRU_INACTIVE_LIST = 1
+        LRU_LIST_NUM = 2
+)
+
+type LruEntry struct {
+	ilist.Entry
+	fr platform.FileRange
+	mr memmap.MappableRange
+	c *CachingInodeOperations
+	accessed bool
+	lru int
```

Comment:
What are accessed and lru? It would be good to have these items documented. I can understand the above, but these are not obvious. Is LRU one of the LRU types above? Why not make it stronger typed?

Reply:
Got it, will be better-documented later. Yes lru is one of the LRU type above, but I'm not sure what is a "stronger type". AFAIK, go does not have enum variables?

Reply:
Go does not have enum variables :( But it does have other types and no implicit casts. Can if you want an enum, you can do:

type LRUStatus int

const (
    LRUActive LRUStatus = iota
    LRUInactive
)

(This is just an example.)

It makes it much harder to make accidental mistakes and also serves to make the code easier to understand.